ament_package/template/isolated_prefix_level/local_setup.bat.in

```diff
@@ -1,53 +1,43 @@
-if not defined AMENT_CURRENT_PREFIX (
-  set "AMENT_CURRENT_PREFIX=@CMAKE_INSTALL_PREFIX@"
-)
-
-:: set type of shell if not already set
-if not defined AMENT_SHELL (
-  set "AMENT_SHELL=bat"
-)
+:: get the directory of this file without trailing backslash
+:: more info about %~dp0 at: http://stackoverflow.com/a/112135
+set "AMENT_CURRENT_PREFIX=%~dp0"
+if %AMENT_CURRENT_PREFIX:~-1%==\ set AMENT_CURRENT_PREFIX=%AMENT_CURRENT_PREFIX:~0,-1%
 
 :: get all packages in topological order
-set "_ORDERED_PACKAGES="
-:ament_get_ordered_packages _ORDERED_PACKAGES "%AMENT_CURRENT_PREFIX%"
+call:ament_get_ordered_packages _ORDERED_PACKAGES "%AMENT_CURRENT_PREFIX%"
 
-:: source prefix-level local_setup.EXT or local_setup.sh file for each package
-:: store AMENT_CURRENT_PREFIX to restore it before each package
-set "_isolated_prefix_local_setup_AMENT_CURRENT_PREFIX=%AMENT_CURRENT_PREFIX%"
-for %%a in ("%_ORDERED_PACKAGES:;=";"%") do (
-  :: restore AMENT_CURRENT_PREFIX for each prefix-level local_setup file
-  set "AMENT_CURRENT_PREFIX=%_isolated_prefix_local_setup_AMENT_CURRENT_PREFIX%\%%a"
-  :: bypass prefix-level local_setup file for performance reasons
-  set "_path=%AMENT_CURRENT_PREFIX%/share/%_package%/local_setup.bat"
-  :: trace output
-  if "%AMENT_TRACE_SETUP_FILES%" NEQ "" echo (call "%_path%")
-  if exist "%_path%" call "%_path%"
-  set "_path="
+:: source prefix-level local_setup.bat file for each package
+if "%_ORDERED_PACKAGES%" NEQ "" (
+  for %%p in ("%_ORDERED_PACKAGES:;=";"%") do (
+    :: bypass prefix-level local_setup file for performance reasons
+    call:call_file "%AMENT_CURRENT_PREFIX%\%%~p\share\%%~p\local_setup.bat"
+  )
+  set "_ORDERED_PACKAGES="
 )
-set "_ORDERED_PACKAGES="
-set "_isolated_prefix_local_setup_AMENT_CURRENT_PREFIX="
 set "AMENT_CURRENT_PREFIX="
 
-:: run any command and arugments passed
-%*
-if %ERRORLEVEL% NEQ 0 exit /b %ERRORLEVEL%
-
-:: prevent double evaluation of function below
-if defined AMENT_TRACE_SETUP_FILES echo Leaving %~0
+:: end of script
 goto:eof
 
+
 :: function to get the packages in this space in topological order
-:: packages are seperated by a semicolon delimiter.
+:: packages are separated by a semicolon delimiter.
 :ament_get_ordered_packages
   setlocal enabledelayedexpansion
   set "ordered_packages="
-  for /f %%a in ('"@PYTHON_EXECUTABLE@" "%AMENT_CURRENT_PREFIX%\_order_isolated_packages.py" "%AMENT_CURRENT_PREFIX%"') do (
-    if "%ordered_packages%" NEQ "" set "ordered_packages=%ordered_packages%;"
-    set "ordered_packages=%ordered_packages%%%a"
+  for /f %%p in ('""@PYTHON_EXECUTABLE@" "%AMENT_CURRENT_PREFIX%\_order_isolated_packages.py" "%~2""') do (
+    if "!ordered_packages!" NEQ "" set "ordered_packages=!ordered_packages!;"
+    set "ordered_packages=!ordered_packages!%%p"
+  )
+  endlocal & (
+    set "%~1=%ordered_packages%"
   )
 goto:eof
+
+:call_file
+  if "%AMENT_TRACE_SETUP_FILES%" NEQ "" echo call "%~1"
+  if exist "%~1" call "%~1%"
```

Comment:
Non of the setup scripts currently prints when files are missing. I kept it the same way in this PR.